common/models.py

```diff
@@ -57,3 +57,14 @@ class UploadFile(TimestampedModel):
         upload_to=get_s3_path
     )
 # END_FEATURE django_storages
+
+
+# START_FEATURE user_action_tracking
+class UserAction(TimestampedModel):
+    user = models.ForeignKey(User, related_name="user_actions", on_delete=models.PROTECT)
+    url = models.URLField(max_length=2083)
+    method = models.CharField(max_length=64)
+    url_name = models.CharField(max_length=256, null=True)
+    status_code = models.IntegerField()
+    user_agent = models.TextField(null=True)
```

Comment:
:art: For `url_name` and `user_agent`, we should mark this as `blank=True` and not `null=True`. Otherwise, we could potentially have both null and empty string values for the field (and in these cases, it seems like they should have the same semantic meaning so we should avoid having both data types be possible).